overlord/devicestate/devicemgr.go

```diff
@@ -141,6 +143,52 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	return m, nil
 }
 
+// StartUp implements StateStarterUp.Startup.
+func (m *DeviceManager) StartUp() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if release.OnClassic {
+		return nil
+	}
+
+	mode := m.SystemMode()
+	if mode != "run" && mode != "recover" {
+		return nil
+	}
+
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if !seeded {
```

Comment:
I don't understand the why of this check?  if we are in run or recover mode with an opened disk we should have the information

Reply:
Should we remove this test altogether and start the snapd-seeding unit in shared keyring mode (in run-snapd-from-snap)? Otherwise what would be an appropriate way to handle the initial seeding without access to the user keyring?